JTS tests and benchmarks

juanignaciosl opened the pull request:
@dkourilov can you review this, please? It's not too much, but I needed to do this before approaching bigger challenges and I wanted to take a look at some tooling :_) In the README you have what you need to run the tests. I managed to test up to 10M points. I need to change my approach for bigger datasets, I was generating it with the python script and PostgreSQL but toe brute force approach doesn't scale.

dkourilov commented on jts/jts-performance/build.sbt:
```diff
@@ -0,0 +1,35 @@
+    name := "jts-performance",
+    version := "0.1",
+    scalaVersion := "2.11.7"
+  )
```
can be replaced with:
`name := ...
version := ...
scalaVersion := ...`
e.g. without `lazy val root =`

juanignaciosl replied:
👍

dkourilov commented:
Can you remove lib/ and replace it with sbt dependency to jts? Also, what jts version are you using?

juanignaciosl commented:
1.14. I just removed the binaries and added the SBT dependencies. I thought that it hadn't been published to a maven repository, but I overlooked that they had splitted the original jts artifact on several libraries.

dkourilov commented on jts/jts-performance/src/test/scala/JTSBenchmark.scala:
```diff
@@ -0,0 +1,44 @@
+    long <- Gen.range("longs")(-179, 179, 5)
+  } yield (new Coordinate(lat, long))
+
+  val simpleIndexedPolygonLocator = new IndexedPointInAreaLocator(simplePolygon)
```
I propose to exclude IndexedPointInAreaLocator from test - this is a second-order optimisation.

juanignaciosl replied:
Ok. Since indexed tests can be useful in the future I've split the code and focused on simple one for benchmarking.

dkourilov commented:
Could you please also include into README:
- jvm used (should be Oracle 64-bit server hotspot)
- jstat numbers during the test (or a jvisualvm screenshot of gc profile)

juanignaciosl commented:
I've added the jvm info to the reporter output (I was using that jvm version) and pasted it at the README.md.

Added both sample of jstat numbers and jvisualvm screenshot.
